app/models/vote.rb

```diff
@@ -0,0 +1,16 @@
+class Vote < ActiveRecord::Base
+  validates :value, :inclusion => {:in => [-1,0,1]}
+  # validates :question, :answer, :comment , uniqueness: {scope: :user, message: "you already voted on this"}
+  belongs_to :user
+  belongs_to :votable, polymorphic: true
+
+  def increment
+    val = self.value
```

Comment:
This val variable doesn't clarify anything. You could do
`update_attributes(value: value + 1)`